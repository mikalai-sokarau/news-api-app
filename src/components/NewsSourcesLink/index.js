import React from 'react';
import { Link } from 'react-router-dom';

const NewsSourcesLink = ({ item, source }) => (
  <Link
    to={{ pathname: `/${item.shortName}` }}
    className={`collection-item ${source === item.shortName ? 'active' : ''}`}
  >
    {item.name}
  </Link>
);

export default NewsSourcesLink;
