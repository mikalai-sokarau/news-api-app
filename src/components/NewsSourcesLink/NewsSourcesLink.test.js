import React from 'react';
import { shallow } from 'enzyme';
import NewsSourcesLink from '../NewsSourcesLink';

describe('<NewsSourcesLink/>', () => {
  test('render a Message', () => {
    const wrapper = shallow(
      <NewsSourcesLink
        item={{
          pathName: '/fakePath',
          shortName: 'fakeData'
        }}
        source="fakeData"
      />
    );

    expect(wrapper).toMatchSnapshot();
  });
});
