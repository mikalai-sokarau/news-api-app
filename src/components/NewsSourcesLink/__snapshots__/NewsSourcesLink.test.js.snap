// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NewsSourcesLink/> render a Message 1`] = `
<Link
  className="collection-item active"
  replace={false}
  to={
    Object {
      "pathname": "/fakeData",
    }
  }
/>
`;
